packages/form-js-viewer/src/render/hooks/useValuesAsync.js

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState } from 'preact/hooks';
+import useService from './useService';
+
+/**
+ * @enum { String }
+ */
+export const LOAD_STATES = {
+  LOADING: 'loading',
+  LOADED: 'loaded',
+  ERROR: 'error'
+};
+
+/**
+ * @typedef {Object} ValuesGetter
+ * @property {Object[]} values - The values data
+ * @property {(LOAD_STATES)} state - The values data's loading state, to use for conditional rendering
+ */
+
+/**
+ * A hook to load values for single and multiselect components.
+ *
+ * @param {Object} field - The form field to handle values for
+ * @return {ValuesGetter} valuesGetter - A values getter object providing loading state and values
+ */
+export default function(field) {
+  const {
+    valuesKey,
+    values: staticValues
+  } = field;
+
+  const [ valuesGetter, setValuesGetter ] = useState({ values: [], state: LOAD_STATES.LOADING });
+  const initialData = useService('form')._getState().initialData;
+
+  useEffect(() => {
+
+    let values = [];
+
+    if (valuesKey !== undefined) {
```

Comment:
Why not simply?

```suggestion
    if (valuesKey) {
```

Reply:
I guess this can be argued, but the presence of the key itself is how we determine what type of source we use. I'm re-enforcing it here, but this is mostly on the front end I suppose so I don't know exactly how justified I am. 

Adding this would make us a tiny bit more redundant (we could have valuesKey=null, values=[someSet] still work), but at the same time we're agreeing that we don't accept two types of sources on the form spec, this is how the rest of the codebase works so I'm not sure.